projects/vdk-plugins/vdk-properties-fs/src/vdk/plugin/properties_fs/fs_properties_client.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import logging
+import os
+from configparser import ConfigParser
+from typing import Dict
+
+from vdk.api.plugin.plugin_input import IPropertiesServiceClient
+
+log = logging.getLogger(__name__)
+
+
+class FileSystemPropertiesServiceClient(IPropertiesServiceClient):
+    """
+    Implementation of IProperties that are kept on local FS only. Stateless.
+    A single properties file store, supporting multiple teams and data jobs properties,
+    via placing the key-values in a dedicated team and job name section to deduplicate.
+    """
+
+    def __init__(self, directory: str, filename: str):
+        self._file_path = os.path.join(directory, filename)
+
+    def read_properties(self, job_name: str, team_name: str) -> Dict:
+        config_parser = ConfigParser()
+        config_parser.read(self._file_path)
+
+        # filter properties by prefix for this particular team and data job
+        prefix = FileSystemPropertiesServiceClient._prefix(team_name, job_name)
+        if config_parser.has_section(prefix):
+            return dict(config_parser.items(prefix))
+        return {}
+
+    def write_properties(self, job_name: str, team_name: str, properties: Dict) -> Dict:
+        log.warning(
+            "You are using local FS Properties client. "
+            "That means the properties will not be stored using the Control Service."
+            "The properties will be read and written to the local file system only."
+        )
+        config_parser = ConfigParser()
+        config_parser.read(self._file_path)
+
+        # filter properties by prefix for this particular team and data job
+        prefix = FileSystemPropertiesServiceClient._prefix(team_name, job_name)
```

Comment:
I think we should keep that as JSON and not as in file. 

ini structure is much more limited - no types support (which properties have) no nested support (properties can be arbitrary nested) . 

And it's pretty easy to implement: `json.dumps(properties)` and `json.loads(properteis)`. 

20 lines become 1 :)

Reply:
I am ok to merge it as it is. But let's create a story/github issue for future improvement

Reply:
I've decided to use JSON, for fully supporting nested data structures.